### ISSUE TITLE
Make `!source` a slash command

### ISSUE THREAD

**Issue by DMFriends** [python-discord/bot#2424]
I think the `!source` command should be made a slash command because, similarly to the reason why we made `!tags` a slash command, it would make it easier to look up the name of a cog without having to leave the Discord channel if you don't exactly remember the name offhand.
Note: I'm fine with someone else implementing this if approved, I don't really have the time right now to contribute at all

**Comment by Robin5605** [comment]
I assume we'd also implement autocomplete here?

**Comment by DMFriends** [comment]
Yes, that's what I was referring to when I said "it would make it easier to look up the name of a cog"

**Comment by mbaruh** [comment]
Sounds fine to me, although the cog would need to go through some structural changes to support auto-complete. Technically it could produce a static mapping of all possible terms and their appropriate info (type, link etc.), since I don't see why it would change without the bot restarting in the process. It would have to be the last cog to be loaded though.

**Comment by Ibrahim2750mi** [comment]
Can I be assigned this?

**Comment by mbaruh** [comment]
Assigned
